src/compiler/checker.ts

```diff
@@ -8063,6 +8067,17 @@ namespace ts {
             return type.resolvedBaseConstructorType;
         }
 
+        function getImplementsTypes(type: InterfaceType): BaseType[] {
+            if (!type.resolvedImplementsTypes) {
+                if (type.symbol.flags & SymbolFlags.Class) {

```

Comment:
Since `type` can be other things than a class based of  `flags`, I thought it best to add the assertion in case someone else uses it in some other context. At the moment I use it only from `serializeAsClass` where the `type` is already guaranteed to be a class by previous checks. 

Should I just remove the check?

Reply:
At the moment, you're basically just checking that `getDeclaredTypeOfClassOrInterface` returns a class -- and nothing in the code itself actually relies on the type being a class. I'd remove it. (And In case `getImplementsTypes` gets used elsewhere, I think usage is pretty obvious.)